### ISSUE TITLE
Rule proposal: no-reserved-component-names

### ISSUE THREAD

**Issue by Mouvedia** [vuejs/eslint-plugin-vue#486]
see https://stackoverflow.com/a/22545622/248058

Those names should produce an error:

- annotation-xml
- color-profile
- font-face
- font-face-src
- font-face-uri
- font-face-format
- font-face-name
- missing-glyph

**Comment by chrisvfritz** [comment]
This makes sense, but I think it should be focused on avoiding reserved names in component definitions, e.g. `no-reserved-component-names` and it would be in `essentials`.

**Comment by chrisvfritz** [comment]
Some examples of code it should catch:

```js
Vue.component('font-face', {})
```

```js
const name = 'font-face'

Vue.component(name, {})
```

```js
// filename: font-face.vue

export default {
  name: 'font-face'
}
```

```js
// filename: font-face.vue

module.exports = {
  name: 'font-face'
}
```

**Comment by michalsnik** [comment]
I think we should also warn about:
```
export default {
  components: {
    'font-face': ...,
  }
}
```

This returns same warning in Vue app, as former examples. @chrisvfritz @shadskii

**Comment by shadskii** [comment]
I've added warnings for locally registered components in #757. Thank you for the suggestion!
